docs/Customization/Custom-Core.rst

```diff
@@ -0,0 +1,295 @@
+.. _custom_core:
+
+Adding a custom core
+====================
+
+You may want to integrate a custom RISC-V core into the Chipyard framework. This documentation page provides step-by-step
+instructions on how to achieve this.
+
+.. note:: 
+
+    RoCC is currently not supported by cores other than Rocket and BOOM. Please use Rocket or BOOM as the RoCC base core if you need to use RoCC.
+
+
+Wrap Verilog Module with Blackbox (Optional)
+--------------------------------------------
+
+Since Chipyard uses Scala and Chisel, if the top module of your core is not in Chisel, you will first need to create a Verilog
+blackbox for it so that it can be processed by Chipyard. See :ref:`incorporating-verilog-blocks` for instructions.
+
+Create Parameter Case Classes
+-----------------------------
+
+Chipyard will generate a core for every ``InstantiableTileParams`` object it discovered in the ``TilesLocated(InSubsystem)`` key.
+This object is derived from``TileParams``, a trait containing the information needed to create a tile. All cores must have
+their own implementation of ``InstantiableTileParams``, as well as ``CoreParams`` which is passed as a field in ``TileParams``.
+
+``TileParams`` holds the parameters for the tile, which include parameters for all components in the tile (e.g.
+core, cache, MMU, etc.), while ``CoreParams`` contains parameters specific to the core on the tile. 
+They must be implemented as case classes with fields that can be overridden by 
+other config fragments as the constructor parameters. See the appendix at the bottom of the page for a list of 
+variable to be implemented. You can also add custom fields to them, but standard fields should always be preferred. 
+
+``InstantiableTileParams[TileType]`` holds the constructor of ``TileType`` on top of the fields of ``TileParams``, 
+where ``TileType`` is the tile class (see the next section).
+All custom cores will also need to implement ``instantiate()`` in their tile parameter class to return a new instance
+of the tile class ``TileType``. 
+
+``TileParams``, ``InstantiableTileParams[TileType]`` and ``CoreParams`` contains the following fields:
```

Comment:
```suggestion
` ``TileParams`` `<https://github.com/chipsalliance/rocket-chip/blob/master/src/main/scala/tile/BaseTile.scala>_, ` ``InstantiableTileParams[TileType]`` `<https://github.com/chipsalliance/rocket-chip/blob/master/src/main/scala/tile/BaseTile.scala>_ and ` ``CoreParams`` `<https://github.com/chipsalliance/rocket-chip/blob/master/src/main/scala/tile/Core.scala>_ contain the following fields:
```

Reply:
I'm not sure I got the syntax exactly right, but this is what I meant by adding the html links. Having the links as a comment in the code example is nice, but would make more sense a a path in the repo. It is much more useful when a user can click on a link in the docs and directly see the updated code on the github web UI

Reply:
It seems to be not possible to keep a link in inline code format while indicating that it is a clickable link (and even making the inline code a link is very tricky). I removed the inline code format from the links here.

Reply:
I also added a note to remind users that there might be discrepancy between the source files in Rocket repo and the the code on the page.